crypto/provider_core.c

```diff
@@ -673,13 +754,53 @@ static int core_thread_start(const OSSL_PROVIDER *prov,
     return ossl_init_thread_start(prov->provctx, handfn);
 }
 
+/*
+ * The FIPS module inner provider doesn't implement these.  They aren't
+ * needed there, since the FIPS module upcalls are always the outer provider
+ * ones.
+ */
+#ifndef FIPS_MODE
+static void core_put_error(const OSSL_PROVIDER *prov,
+                           uint32_t reason, const char *file, int line)
+{
+    /*
+     * If the uppermost 8 bits are non-zero, it's an OpenSSL library
+     * error and will be treated as such.  Otherwise, it's a new style
+     * provider error and will be treated as such.
+     */
+    if (ERR_GET_LIB(reason) != 0) {
```

Comment:
Again, magic numbers are bad. If you're assigning special semantics to 0 as a LIB code it should be documented (I think) and have a define.

Reply:
If you look at `crypto/err/err.c`, you will see this "magic number" all over the place.  A macro for it should end up in `include/openssl/err.h`, and since that implies a modification of the ERR subsystem, I would suggest such a change belongs on another PR for that topic.  Make a special macro for *this* use without involving the ERR subsystem as a whole is just as bad as using the "magic number" directly.

Reply:
okay